src/main/scala/com/microsoft/hyperspace/index/dataskipping/DataSkippingIndexConfig.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (2021) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index.dataskipping
+
+import org.apache.spark.sql.DataFrame
+
+import com.microsoft.hyperspace.HyperspaceException
+import com.microsoft.hyperspace.index.{IndexConfigTrait, IndexerContext}
+import com.microsoft.hyperspace.index.dataskipping.sketch.Sketch
+
+case class DataSkippingIndexConfig(
```

Comment:
Please add scaladoc for this new public API. It is very important to understand how to properly use this new kind of index. Please respond to the following questions: 

1. What's a `sketch` in the context of this index?
2. What's a `sketchOffset` and how can it be used?
3. Is it a 1-1 mapping between a `sketch` and a column?